divi/src/txmempool.cpp

```diff
@@ -703,7 +713,8 @@ void CTxMemPool::check(const CCoinsViewCache* pcoins) const
         for (const auto& txin : tx.vin) {
             // Check that every mempool transaction's inputs refer to available coins, or other mempool tx's.
             CTransaction tx2;
-            if (lookup(txin.prevout.hash, tx2)) {
+            if ((segwitLight && lookupBareTxid(txin.prevout.hash, tx2))
+                  || (!segwitLight && lookup(txin.prevout.hash, tx2))) {
```

Comment:
In the event of timestamps not being monotonic, a transaction that is saved in one block under 'segwit-light' will be 'not found' if the next block is not 'segwit-light'

Reply:
I can't find this place in the code anymore, but I guess what you mean here is that the mempool logic can be inconsistent around fork activation.  This is the case independent of timestamps being monotonic or not, and is why we have this special rule of forbidding spending of unconfirmed outputs in the mempool policy (and wallet) for a time window around the fork.  This time window should be large enough to make sure there are no issues like this in practice.

On the core consensus level, non-monotonic timestamps are of course fine as well.  For a transaction confirmed in a particular block, that block's timestamp (independent of what came before and comes after) determines what UTXOs get created, and the wallet later also handles this correctly when spending those UTXOs again.